Fix #4705. Install modules in SvelteKit

slava-viktorov opened the pull request:
Add a new "modules" property to the Svelte component. To solve the problem with installing additional Swiper modules when used in SvelteKit.

Fix #4705

hmaesta commented:
Am I missing something here? Updated to 6.8.0 and couldn't make it work.
https://codesandbox.io/s/sveltekit-swiper-3dxrg?file=/src/routes/index.svelte

slava-viktorov commented:
> Am I missing something here? Updated to 6.8.0 and couldn't make it work.
> https://codesandbox.io/s/sveltekit-swiper-3dxrg?file=/src/routes/index.svelte
Please look at the pull request and the changes made to the code. I added a new property for the <Swiper>component
https://codesandbox.io/s/sveltekit-swiper-forked-iin8p?file=/src/routes/index.svelte

hmaesta commented:
Oh, now  I get it. Thanks, Slava! 🤝

nolimits4web commented:
@hmaesta also check the updated docs https://swiperjs.com/svelte#using-with-svelte-kit , it has information about new  `modules` prop

ProbeConstructor commented:
hi, im new here and i want to learn why this works: https://codesandbox.io/s/h91ee
but this does not : https://codesandbox.io/s/5jix6
can someone guide me on this please, thank you

mweissen commented:
@ProbeConstructor: I would like to know that too.

@slava: which new component was added? I haven't worked with the codesandbox enough yet. I am not clear where to find the pull request and what was really changed with it. Are the changes now part of swiperjs for sveltekit? Or in other words, how do I currently get swiper to work with sveltekit?

slava-viktorov commented:
@ProbeConstructor @mweissen 
Just read the documentation if you can't see the pull request and understand the changes in the code. The link to the documentation is in the previous comment. I will insert it here again for you, because it looks like this is a big deal for you https://swiperjs.com/svelte#using-with-svelte-kit.
But if you can't do that either, then just UPDATE sweeper to the LATEST VERSION and copy the code from https://codesandbox.io/s/sveltekit-swiper-forked-iin8p?file=/src/routes/index.svelte

hmaesta commented:
A Codesandbox for anyone looking for **SvelteKit** + **Swiper 7**:
- https://codesandbox.io/s/sveltekit-swiper-7-coq4t?file=/src/routes/index.svelte
